corehq/apps/dump_reload/sql/load.py

```diff
@@ -18,6 +22,17 @@
 from corehq.apps.dump_reload.util import get_model_label
 from corehq.sql_db.routers import HINT_PARTITION_VALUE
 
+CHUNK_SIZE = 200
+
+
+class DefaultDictWithKey(defaultdict):
+    """
+    A defaultdict that passes ``key`` to its factory
+    """
+    def __missing__(self, key):
+        self[key] = value = self.default_factory(key)
+        return value
```

Comment:
Could move to bottom of file since this is a utility class only used by code below (newspaper metaphor).